l10n_it_fatturapa_out/tests/test_fatturapa_xml_validation.py

```diff
@@ -886,6 +888,94 @@ def test_no_tax_fail(self):
         )
         self.assertEqual(ue.exception.args[0], error_message)
 
+    def test_partner_no_address_fail(self):
+        """
+        - create an XML invoice where the customer has no address or city
+
+        expect to fail with a proper message
+        """
+        self.res_partner_fatturapa_0.street = False
+        self.res_partner_fatturapa_0.city = False
+        invoice = self.invoice_model.create(
+            {
+                "invoice_date": "2016-06-15",
+                "partner_id": self.res_partner_fatturapa_0.id,
+                "journal_id": self.sales_journal.id,
+                "invoice_payment_term_id": self.account_payment_term.id,
+                "user_id": self.user_demo.id,
+                "move_type": "out_invoice",
+                "currency_id": self.EUR.id,
+                "invoice_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "account_id": self.a_sale.id,
+                            "product_id": self.product_product_10.id,
+                            "name": "Mouse, Optical",
+                            "quantity": 1,
+                            "product_uom_id": self.product_uom_unit.id,
+                            "price_unit": 10,
+                            "tax_ids": [(6, 0, {self.tax_22.id})],
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "account_id": self.a_sale.id,
+                            "product_id": self.product_order_01.id,
+                            "name": "Zed+ Antivirus",
+                            "quantity": 1,
+                            "product_uom_id": self.product_uom_unit.id,
+                            "price_unit": 4,
+                            "tax_ids": [(6, 0, {self.tax_22.id})],
+                        },
+                    ),
+                ],
+            }
+        )
```

Comment:
Potrebbe essere un filino più sintetico, visto che non ti servono esattamente tutti questi dati per il test
```suggestion
        invoice = self._create_invoice()
```
e poi modifichi `invoice.partner_id.street/city`.